### ISSUE TITLE
Crash on Arch Linux

### ISSUE THREAD

**Issue by Aloso** [ArturKovacs/emulsion#16]
The window appears for a fraction of a second and even displays the image correctly, then crashes with the following console output:

```sh
Panic happened

--
Too many mipmap levels, received 4, maximum for this texture dimension is 3.
--

Location /home/ludwig/.cargo/registry/src/github.com-1ecc6299db9ec823/glium-0.27.0/src/texture/mod.rs:248:21

   0: emulsion::handle_panic::handle_panic
   1: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:515
   2: rust_begin_unwind
             at src/libstd/panicking.rs:419
   3: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:373
   4: glium::texture::MipmapsOption::num_levels
   5: glium::texture::any::new_texture
   6: emulsion::image_cache::image_loader::texture_from_image
   7: emulsion::image_cache::ImageCache::load_specific
   8: emulsion::playback_manager::PlaybackManager::update_image
   9: <emulsion::picture_widget::PictureWidget as gelatin::Widget>::before_draw
  10: <gelatin::line_layout_container::LineLayoutContainer<Dim> as gelatin::Widget>::before_draw
  11: <gelatin::line_layout_container::LineLayoutContainer<Dim> as gelatin::Widget>::before_draw
  12: gelatin::window::Window::redraw
  13: gelatin::application::Application::start_event_loop::{{closure}}
  14: winit::platform_impl::platform::x11::EventLoop<T>::run
  15: winit::platform_impl::platform::EventLoop<T>::run
  16: winit::event_loop::EventLoop<T>::run
  17: gelatin::application::Application::start_event_loop
  18: emulsion::main
  19: std::rt::lang_start::{{closure}}
  20: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
      std::panicking::try::do_call
             at src/libstd/panicking.rs:331
      std::panicking::try
             at src/libstd/panicking.rs:274
      std::panic::catch_unwind
             at src/libstd/panic.rs:394
      std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  21: main
  22: __libc_start_main
  23: _start
```

<details>
<summary>My system information</summary>

```sh
 ██████████████████  ████████     ludwig@ludwig-pc
 ██████████████████  ████████     OS: Manjaro 20.0 Lysia
 ██████████████████  ████████     Kernel: x86_64 Linux 5.5.18-3-MANJARO
 ██████████████████  ████████     Uptime: 8h 4m
 ████████            ████████     Packages: 1653
 ████████  ████████  ████████     Shell: nu
 ████████  ████████  ████████     Resolution: 3200x900
 ████████  ████████  ████████     DE: KDE 5.69.0 / Plasma 5.18.4
 ████████  ████████  ████████     WM: KWin
 ████████  ████████  ████████     GTK Theme: Material-Black-Blueberry [GTK2/3]
 ████████  ████████  ████████     Icon Theme: breeze-dark
 ████████  ████████  ████████     Disk: 677G / 1,1T (70%)
 ████████  ████████  ████████     CPU: AMD A8-7600 Radeon R7, 10 Compute Cores 4C+6G @ 4x 3.1GHz
 ████████  ████████  ████████     GPU: AMD KAVERI (DRM 2.50.0, 5.5.18-3-MANJARO, LLVM 9.0.1)
                                  RAM: 7236MiB / 14925MiB
```

</details>

**Comment by ArturKovacs** [comment]
I'm guessing this is a GPU driver limitation. My plan to solve this is to generate all mipmaps levels if the functions call fails.

**Comment by ArturKovacs** [comment]
I made a fix. Please check out the [`fix-mipmaps`](https://github.com/ArturKovacs/emulsion/tree/fix-mipmaps) branch and see if that work with the image that crashed it earlier.

**Comment by omac777** [comment]
I'll get to testing the fix-mipmaps shortly.

on archlinux here is the output:
```
glxinfo | grep -i opengl
OpenGL vendor string: X.Org
OpenGL renderer string: AMD Radeon (TM) R9 380 Series (TONGA, DRM 3.36.0, 5.6.6-arch1-1, LLVM 10.0.0)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 20.0.4
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.6 (Compatibility Profile) Mesa 20.0.4
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 20.0.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
OpenGL ES profile extensions:
```

On archlinux, there is no such package as mesa-utils, but I did my best to install the appropriate mesa packages as listed with those "[installed]".

```
pacman -Ss mesa
extra/glu 9.0.1-1 [installed]
    Mesa OpenGL Utility library
extra/libva-mesa-driver 20.0.4-2 [installed]
    VA-API implementation for gallium
extra/mesa 20.0.4-2 [installed]
    An open-source implementation of the OpenGL specification
extra/mesa-demos 8.4.0-3 [installed]
    Mesa demos and tools incl. glxinfo + glxgears
extra/mesa-vdpau 20.0.4-2
    Mesa VDPAU drivers
extra/opencl-mesa 20.0.4-2 [installed]
    OpenCL support for AMD/ATI Radeon mesa drivers
extra/vulkan-intel 20.0.4-2
    Intel's Vulkan mesa driver
extra/vulkan-mesa-layer 20.0.4-2 [installed]
    Vulkan overlay layer to display information about the application
extra/vulkan-radeon 20.0.4-2 [installed]
    Radeon's Vulkan mesa driver
community/adriconf 1.6.1-1
    GUI tool to configure Mesa drivers by setting options and writing them to the standard drirc file
community/i810-dri 7.11.2-10
    Mesa DRI drivers for Intel i810
community/mach64-dri 7.11.2-10
    Mesa DRI drivers for ATI Mach64
community/mga-dri 7.11.2-10
    Mesa DRI drivers for Matrox
community/r128-dri 7.11.2-10
    Mesa DRI drivers for ATI Rage128
community/savage-dri 7.11.2-10
    Mesa DRI drivers for S3 Sraphics/VIA Savage
community/sis-dri 7.11.2-10
    Mesa DRI drivers for SiS
community/tdfx-dri 7.11.2-10
    Mesa DRI drivers for 3dfx
community/unichrome-dri 7.11.2-10
    Mesa DRI drivers for S3 Graphics/VIA Unichrome
multilib/lib32-glu 9.0.1-1
    Mesa OpenGL utility library (32 bits)
multilib/lib32-libva-mesa-driver 20.0.4-2
    VA-API implementation for gallium (32-bit)
multilib/lib32-mesa 20.0.4-2 [installed]
    An open-source implementation of the OpenGL specification (32-bit)
multilib/lib32-mesa-demos 8.4.0-1
    Mesa demos and tools (32-bit)
multilib/lib32-mesa-vdpau 20.0.4-2
    Mesa VDPAU drivers (32-bit)
multilib/lib32-opencl-mesa 20.0.4-2
    OpenCL support for AMD/ATI Radeon mesa drivers (32-bit)
multilib/lib32-vulkan-intel 20.0.4-2
    Intel's Vulkan mesa driver (32-bit)
multilib/lib32-vulkan-radeon 20.0.4-2
    Radeon's Vulkan mesa driver (32-bit)
```

On archlinux, when I invoke the similar, it still hangs and doesn't display the first image:
```
git clone https://github.com/ArturKovacs/emulsion.git
cd emulsion/
cargo build --release
./target/release/emulsion /home/blah/Pictures/Screenshot\ from\ 2019-04-25\ 06-15-37.png
```
be right back.  Will checkout fix-mipmaps.

**Comment by omac777** [comment]
```
cargo clean
git checkout fix-mipmaps
cargo build --release
sync
sync
./target/release/emulsion /home/david/Pictures/Screenshot\ from\ 2019-04-25\ 06-15-37.png 
```

No success, still hangs with "Quit or Wait" dialog.

I don't need to remind you I did a "pacman -Syuu" before starting all this.

**Comment by omac777** [comment]
I also tried installing these two packages:
community/adriconf
extra/mesa-vdpau

It didn't help.  adriconf is a nice gui for adding different gpu diagnostic config stuff, but I'm not at that level of proficiency with gpu stuff.

For your information when I run glxgears, it behaves and displays correctly.

**Comment by ArturKovacs** [comment]
>
> No success, still hangs with "Quit or Wait" dialog.

@omac777  All the information that you kindly provided don't seen to be related to this issue. Please read the very first comment carefully before adding a new comment to an issue. See if the error that you are experiencing is the same as the one described in the first comment. If it's not the same, than any information about your particular case is not related to this one and just adds clutter.

**Comment by Aloso** [comment]
@ArturKovacs Thank you, the problem is fixed in the `fix-mipmaps` branch. This can be closed.

@omac777 Would it be okay for you to continue this discussion in another issue?

**Comment by ArturKovacs** [comment]
I'm closing this as the issue was solved by 9770443fc47be39c83ba3ba5514162dc339fb13b.

@omac777 Please consider moving the information you provided to a new issue,
